pkg/gpu/nvidia/alpha_plugin.go

```diff
@@ -33,22 +33,17 @@ type pluginServiceV1Alpha struct {
 
 func (s *pluginServiceV1Alpha) ListAndWatch(emtpy *pluginapi.Empty, stream pluginapi.DevicePlugin_ListAndWatchServer) error {
 	glog.Infoln("device-plugin: ListAndWatch start")
-	changed := true
+	if err := s.sendDevices(stream); err != nil {
+		return err
+	}
 	for {
-		if changed {
-			resp := new(pluginapi.ListAndWatchResponse)
-			for _, dev := range s.ngm.devices {
-				resp.Devices = append(resp.Devices, &pluginapi.Device{ID: dev.ID, Health: dev.Health})
-			}
-			glog.Infof("ListAndWatch: send devices %v\n", resp)
-			if err := stream.Send(resp); err != nil {
-				glog.Errorf("device-plugin: cannot update device states: %v\n", err)
-				s.ngm.grpcServer.Stop()
+		select {
+		case d := <-s.ngm.health:
```

Comment:
The channel should never be closed (no one will close it), and it will only have values when a device goes unhealthy, so there should not be any unnecessary `sendDevices` calls.